library/common-library/templates/_images.tpl

```diff
@@ -7,7 +7,7 @@ Return the proper image name
     {{- $repositoryName := include "common.images.repository" .imageRoot -}}
     {{- $tag := include "common.images.tag" ( dict "imageRoot" .imageRoot "context" .context) -}}
 
-    {{- if $registryName }}
+    {{- if $registryName -}}
```

Comment:
just out of curiosity. What is the effect of the this trim operator on conditional entries?

Reply:
Exactly the same as other templates. Trim white spaces to the left. Here is an example:
```mustache
apiVersion: v1
kind: ConfigMap
metadata:
  name: test
data:
  key: |
    1
    {{ if include "newrelic.common.verboseLog" . }}
    {{ end }}

    2
    {{ if (include "newrelic.common.nrStaging" . ) }}
    {{ end }}

    3
    {{ with include "newrelic.common.proxy" . }}
    {{ end }}
```
outputs:
```yaml
# Source: CHART-TEMPLATE/templates/test.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: test
data:
  key: |
    1


    2


    3
```
while with line trimmers
```mustache
apiVersion: v1
kind: ConfigMap
metadata:
  name: test
data:
  key: |
    1
    {{- if include "newrelic.common.verboseLog" . }}
    {{- end }}

    2
    {{- if (include "newrelic.common.nrStaging" . ) }}
    {{- end }}

    3
    {{- with include "newrelic.common.proxy" . }}
    {{- end }}

```
outputs
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: test
data:
  key: |
    1

    2

    3
```
If you ask explicitly why to trim in the `if` is because the `if` one is going to create a new line that rarely be seen because it could be eaten by the `- end` but could be led in some cases to an empty line depending what is inside the if.

Reply:
```mustache
apiVersion: v1
kind: ConfigMap
metadata:
  name: test
data:
  key: |
    1
    {{- if include "newrelic.common.verboseLog" . }}
    {{ end }}

    2
    {{- if (include "newrelic.common.nrStaging" . ) }}
    {{ end }}

    3
    {{- with include "newrelic.common.proxy" . }}
    {{ end }}
```
outputs also
```yaml
---
# Source: CHART-TEMPLATE/templates/test.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: test
data:
  key: |
    1

    2

    3
```